src/variable.cpp

```diff
@@ -118,4 +131,42 @@ std::string Variable::getInitialValue() const
     return mPimpl->mInitialValue;
 }
 
+void Variable::setInterfaceType(const Variable::INTERFACE_TYPES interfaceType)
+{
+    mPimpl->mInterfaceType = interfaceType;
+}
+
+Variable::INTERFACE_TYPES Variable::getInterfaceType() const
+{
+    return mPimpl->mInterfaceType;
+}
+
+std::string interfaceTypeToString(Variable::INTERFACE_TYPES interfaceType)
+{
+    std::string str;
+    switch (interfaceType) {
+    case Variable::INTERFACE_TYPES::INTERFACE_TYPE_UNDEFINED: {
+        str = "";
```

Comment:
I seem to be dropping a bit of coverage here since the if statement leading in to `interfaceTypeToString()` excludes the undefined interfaces since we don't want to serialise those.

I'm guessing there's possibly a cleverer C++11 way to do this? If so, I bet @hsorby knows what it is :-)